src/otp_workflow.js

```diff
@@ -0,0 +1,129 @@
+//docker run -i --rm -v $(pwd)/src:/src  -e FUNC_KEY=${FUNC_KEY} -e FUNC_MERCHANT_KEY=${FUNC_MERCHANT_KEY} -e BASE_URL=${BASE_URL} -e BASE_MERCHANT_URL=${BASE_MERCHANT_URL} loadimpact/k6 run /src/otp_workflow.js
+
+import http from 'k6/http';
+import { sleep } from 'k6';
+import { check } from 'k6';
+
+export let options = {
+    vus: 1,
+    iterations: 20,
+    thresholds: {
+        http_req_duration: ['p(99)<1500'], // 99% of requests must complete below 1.5s
+        'http_req_duration{pagoPaMethod:OtpGeneration}': ['p(95)<1000'], // threshold on API requests only
+        'http_req_duration{pagoPaMethod:OtpCheckNoInvalidate}': ['p(95)<1000'], // threshold on API requests only
+        'http_req_duration{pagoPaMethod:OtpCheckInvalidate}': ['p(95)<1000'],
+        'http_req_duration{pagoPaMethod:CacheMiss}': ['p(95)<1000'],
+        'http_req_duration{pagoPaMethod:InvalidOtp}': ['p(95)<1000'],
+    },
+};
+
+function otpGeneration(fiscalCode, url, funcKey) {
+
+    var payload = JSON.stringify({});
+    var tag = {
+        pagoPaMethod: "OtpGeneration",
+    };
+    var headersParams = {
+        headers: {
+            'Content-Type': 'application/json',
+            'x-functions-key': funcKey
+        },
+    };
+    var r = http.post(url, payload, headersParams, {
+        tags: tag,
+    });
+    console.log("OTP generation for " + fiscalCode + ". Status " + r.status);
+    check(r, { 'status is 200': (r) => r.status === 200 }, tag);
+    return JSON.parse(r.body).code;
```

Comment:
please check if _body_ is undefined first. In case the response is not as you might expect, _body_ could be undefined.

Refer to https://k6.io/docs/testing-guides/running-large-tests/#make-your-test-code-resilient

Reply:
Done!